### ISSUE TITLE
CodiMD prints lots of error messages with invalid postgres config

### ISSUE THREAD

**Issue by dalcde** [codimd/server#408]
Steps to reproduce:
1. Install CodiMD
2. Set `CMD_DB_URL` to a postgres database, but with incorrect password
3. Run `node app.js`
4. The server panics because it fails to connect to the database.
5. Ctrl-C to kill the process
6. The process keeps printing lines of the form
```
2020-06-16T10:31:40.143Z error:         password authentication failed for user "codimd"
```
around 20 such messages per second, non-stop.

**Comment by dalcde** [comment]
I believe the culprit is https://github.com/codimd/server/blob/release/2.0.x/src/lib/utils/functions.ts#L73 . When the process receives a SIGTERM, it tries to save existing changes to database. It retries every 100 milliseconds until it succeeds, and prints the error message otherwise. If the database is not configured properly, this will always fail and keep printing errors. Sending another SIGTERM calls the function again, so we get twice the amount of error messages

I think we should
1. Log explicitly that we are trying to write existing data to the database
2. If we receive another SIGTERM (or any kill signal), we kill the whole process immediately.
3. Lower the retry rate to, say, 1 second.

**Comment by SISheogorath** [comment]
I don't think we should kill the whole process immediately. The proper way would be:

1. Log our intentions: "Will try to write state within the next 30 seconds and terminate"
2. Set up a timeout to terminate in a failed state (`setTimeout(() => {process.exit(1)}, 30)`)
3. Try to write the remaining data to the database and terminate with exit 0 when successfully written

Either we make it to write things, or we terminate after a determined amount of time. Everything else is already handled on the OS level using `SIGKILL`. `SIGTERM` explicitly asks the program to shutdown properly and we can not shutdown properly until the state is written. But I agree we should terminate at some point.

**Comment by dalcde** [comment]
Good point. In any case the second SIGTERM should do nothing instead of
trying to save state again?

**Comment by davidmehren** [comment]
Fix merged into the 2.0 branch.
